### PR TITLE
os/arch/arm/src/amebad: Fix mixed uart log issue

### PR DESCRIPTION
**Pull request by AdamZhang0124**
This PR fixes the mixed uart log issue while multiple functions use log_uart at the same time.

### CODE REVIEW

**Review by jeongchanKim — COMMENTED** [review]
Could you add the commit message what is the change about?

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 1]
```diff
@@ -114,7 +114,7 @@
 #elif defined(CONFIG_UART1_SERIAL_CONSOLE)
 #define CONSOLE_DEV             g_uart1port             /* UART1 is console */
 #define TTYS0_DEV               g_uart1port             /* UART1 is ttyS0 */
-#define CONSOLE                 UART1_DEV
+#define CONSOLE                 UART3_DEV
```
Comment (jeongchanKim) [inline os/arch/arm/src/amebad/amebad_serial.c:117]: All other defines are related to uart1, but this define is uart3.
Is it correct? And if then, why we should set like this?

**Comment by jeongarmy** [comment]
@AdamZhang0124 Could you update this commit with rebase?

File: os/arch/arm/src/amebad/amebad_serial.c (hunk 1 above)
Comment (sunghan-chang) [reply os/arch/arm/src/amebad/amebad_serial.c:117]: What we tested does not include this change. Could you let me know why this is suddenly added?

File: os/arch/arm/src/amebad/amebad_serial.c [hunk 2]
```diff
@@ -672,7 +672,27 @@ static bool rtl8721d_up_txready(struct uart_dev_s *dev)
...
+	}
+	else
+#elif defined(CONFIG_UART2_SERIAL_CONSOLE)
+	if (uart_index_get(priv->tx) == 2){
+		while (!serial_writable(sdrv[uart_index_get(priv->tx)]));
+		return true;
+	}
+	else
+#endif
+		return (serial_writable(sdrv[uart_index_get(priv->tx)]));
```
Comment (sunghan-chang) [inline os/arch/arm/src/amebad/amebad_serial.c:695]: To make it align, let's do as below.
```c
{
      return (...);
}
```

File: os/arch/arm/src/amebad/amebad_serial.c (hunk 1 above)
Comment (jeongchanKim) [reply os/arch/arm/src/amebad/amebad_serial.c:117]: @sunghan-chang In our test code, I missed this change....
@AdamZhang0124 , Anyway, in our test, we can successfully run without this change.
Is this needed?

Comment (AdamZhang0124) [reply os/arch/arm/src/amebad/amebad_serial.c:117]: Hi @sunghan-chang  @jeongchanKim,
This is an notation mismatch I recently found.
The TizenRT uart1 actually maps to Ameba's uart3.
It won't affect the function.

Comment (jeongchanKim) [reply os/arch/arm/src/amebad/amebad_serial.c:117]: Thanks for checking :)

**Comment by jeongarmy** [comment]
@AdamZhang0124 Could you update this commit with rebase master?

**Comment by AdamZhang0124** [comment]
Hi @sunghan-chang @jeongchanKim @jeongarmy ,
Please allow me to close this PR and raise another one, all the discussed questions will be fixed in the new PR, thanks!

**Pull request closed without merging**